src/dataset.py

```diff
@@ -113,7 +126,32 @@ def __iter__(self):
 
         for index, item in enumerate(self.data):
             if index % (num_workers * world_size) == (process_rank * num_workers + worker_id):
-                item = item[1].values[0].copy()
-                x = item[:self.seq_len]
-                y_true = item[1:self.seq_len+1]
+                item = item[1].values[0].tolist()
+                if len(item) <= self.seq_len:
+                    length = len(item)
+                    item = item + [self.pad_token_id]
+                    x = item[:length]
+                    y_true = item[1:length+1]  
+                else:
+                    x = item[:self.seq_len]
+                    y_true = item[1:self.seq_len+1]
                 yield(x,y_true)
+
+    def pad_to_longest(self, batch):
+        """
+        Collator function for padding sequences to longest in batch, during training
+        """
+        x, y_true = zip(*batch)
+
+        x_lengths = [len(line) for line in x]
+        max_x_length = max(x_lengths)
+        padded_x = [line + [self.pad_token_id] * (max_x_length - len(line)) for line in x]
+
+        y_true_lengths = [len(s) for s in y_true]
+        max_y_true_lengths = max(y_true_lengths)
+        padded_y_true = [line + [self.pad_token_id] * (max_y_true_lengths - len(line)) for line in y_true]
```

Comment:
perhaps call padded_y_true "y_padded_true" to have the name help imply padding is on the end.